src/Aggregator.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Algolia\SearchBundle;
+
+use Algolia\SearchBundle\Exception\EntityNotFoundInObjectID;
+use Algolia\SearchBundle\Exception\InvalidEntityForAggregator;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Serializer\Normalizer\NormalizableInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * @ORM\MappedSuperclass
+ */
+abstract class Aggregator implements NormalizableInterface
```

Comment:
I would suggest adding an `AggregatorInterface` and type hinting against that in all the places where you currently check for the `Aggregator`. It makes sense providing a default implementation, but providing an interface along with it will allow people to use their own implementations with custom logic if they want/need it.

Reply:
That's a great suggestion, I actually had an interface before, but has been removed here: https://github.com/algolia/search-bundle/pull/257/commits/a7c86634336aa7855fc9f8bcacd6b4d6d12965ce.

**Reason**: Releasing this feature with an interface now will couple this feature to a contract that can't change without a major version. First, I would prefer to understand if people will have the need of creating their own implementations - if so, we create an interface of it.

Reply:
That makes sense - just wanted to make sure it’s on your mind! 👍